src/curl/http_client.cpp

```diff
@@ -309,4 +315,19 @@ namespace ppconsul { namespace curl {
             throwCurlError(err, m_errBuffer);
     }
 
+    void CurlHttpClient::setHeaders(const http::RequestHeaders & headers)
+    {
+        curl_slist * slist = nullptr;
+        for (const auto & header : headers)
+        {
+            auto record = header.first + ": " + header.second;
```

Comment:
This makes three allocations at worst. Please use `reserve` to have a single allocation.